src/west/commands/project.py

```diff
@@ -491,31 +491,45 @@ def _projects(args, listed_must_be_cloned=True):
         # No projects specified. Return all projects.
         return projects
 
-    # Got a list of projects on the command line. First, check that they exist
-    # in the manifest.
-
-    project_names = [project.name for project in projects]
-    nonexistent = set(args.projects) - set(project_names)
-    if nonexistent:
-        log.die('Unknown project{} {} (available projects: {})'
-                .format('s' if len(nonexistent) > 1 else '',
-                        ', '.join(nonexistent),
-                        ', '.join(project_names)))
-
-    # Return the projects in the order they were listed
+    # Listed but missing projects. Used for error reporting.
+    missing_projects = []
+
     res = []
-    for name in args.projects:
+    for project_arg in args.projects:
         for project in projects:
-            if project.name == name:
+            if project.name == project_arg:
+                # The argument is a project name
                 res.append(project)
                 break
+        else:
+            # The argument is not a project name. See if it is a project
+            # (sub)path.
+            for project in projects:
+                # The startswith() means we also detect subdirectories of
+                # project repositories. Giving a plain file in the repo will
+                # work here too, but that probably doesn't hurt.
+                if os.path.realpath(project_arg).startswith(
+                    os.path.realpath(project.abspath)):
```

Comment:
Hrm, why take realpath of project.abspath? The value in abspath should be as normalized as possible; if we need to eliminate symbolic links in project abspaths, that should be done at manifest parsing time (and in _special_project). The rest of the code should be able to trust that a Project they get handed has been canonicalized.

Reply:
Hmm... yeah, I guess it could be normalized earlier.